scripts/app.js

```diff
@@ -0,0 +1,304 @@
+        console.log("error generating patch: ", error);
+        reject(error);
+      } else {
+        fs.readFile("myPatch.patch", "utf8", function(err, data) {
+          if (err) {
+            console.log("error reading patch file: ", err);
+            reject(err);
+          }
+          resolve(data);
+        });
+      }
+    });
+  })
+}
+async function getGitPatchFromCommits(firstCommit, lastCommit) {
+  //need to delete patch file?
+  return new Promise((resolve, reject) => {
+    if (lastCommit === null) {
+      let patchCommand = "git show HEAD > myPatch.patch";
+      exec(patchCommand, function(error, stdout, stderr) {
+        if (error !== null) {
+          console.log("error generating patch: ", error);
+          reject(error);
+        } else {
+          fs.readFile("myPatch.patch", "utf8", function(err, data) {
+            if (err) {
+              console.log("error reading patch file", err);
+              reject(err);
+            }
+            resolve(data);
+          });
+        }
+      });
+    } else {
+      let patchCommand =
+        "git diff " + firstCommit + "^..." + lastCommit + " > myPatch.patch";
+      exec(patchCommand, function(error, stdout, stderr) {
+        if (error !== null) {
+          console.log("error generating patch: ", error);
+          reject(error);
+        } else {
+          fs.readFile("myPatch.patch", "utf8", function(err, data) {
+            if (err) {
+              console.log("error reading patch file ", err);
+              reject(err);
+            }
+            resolve(data);
+          });
+        }
+      });
+    }
+  });
+}
+
+async function main() {
+  //world or repo build is passed in through cmd line/makefile
+  const buildSize = process.argv[2];
+  const patchFlag = process.argv[3];
+  const userName = await getGitUser();
+  const userEmail = await getGitEmail();
+  const url = await getRepoInfo();
+  const repoName = getRepoName(url);
+  const branchName = await getBranchName();
+  const newHead = "genericscsss";
+  // toggle btwn create patch from commits or what you have saved locally
+  if (patchFlag === "commit") {
+    const { firstCommit, lastCommit } = await getGitCommits();
+    const patch = await getGitPatchFromCommits(firstCommit, lastCommit);
+    const payLoad = await createPayload(
+      repoName,
+      branchName,
+      userName,
+      url,
+      patch,
+      buildSize,
+      newHead
+    );
+    const success = insertJob(
+      payLoad,
+      "Github Push: " + userName + "/" + repoName,
+      userName,
+      userEmail
+    );
+  }
+
+  if(patchFlag === "local"){
+    const patch = await getGitPatchFromLocal();
+    const payLoad = await createPayload(
+      repoName,
+      branchName,
+      userName,
+      url,
+      patch,
+      buildSize,
+      newHead
+    );
+    const success = insertJob(
+      payLoad,
+      "Github Push: " + userName + "/" + repoName,
+      userName,
+      userEmail
+    );   
+
+  }
+
```

Comment:
check for success on record insert and report and error to the console on failure.

Reply:
Inside insertJob, I do error handling of record insert. Do you want me to also check for error/success inside main?

Reply:
I put logs inside insertJob